kubernetes/provider_test.go

```diff
@@ -214,6 +215,18 @@ func skipIfNoAwsSettingsFound(t *testing.T) {
 	}
 }
 
+func skipIfNoAzureCloudSettingsFound(t *testing.T) {
+	if os.Getenv("TF_VAR_aks_client_id") == "" && os.Getenv("ARM_CLIENT_ID") == "" {
```

Comment:
The other skip functions in the tests don't use the `TF_VAR_` convention - why have we decided to use it for azure?

Reply:
It was included in the Azure PR, and I remember testing it and thinking it was necessary. But I'll double check to be sure.

Reply:
My 2 cents here is to move away entirely from checking environment type based on credentials being present. It's not as precise as checking node labels, which we already do for some cluster types. IMO we should move towards using node labels across the board.

The reason why I'm advising against it is because they are easily producing false positives in runtime environments where more than one set of credentials is present (like cloning a TC config for an existing cloud provider and forgetting to remove the old credentials or more commonly local runtime environments where we might have all creds available).

Reply:
That sounds good to me. I'll remove it from this PR and do the work in the Azure PR. I already put a node check in that branch anyway. Thanks!